### PR TITLE
test: Fix syncwithvalidationinterfacequeue calls

### LINKED ISSUE #26071: test: failure in wallet_resendwallettransactions.py
https://cirrus-ci.com/task/5603470202896384?logs=ci#L2681

### PR DESCRIPTION
**Pull request by maflcko**
Fixes #26071

### CODE REVIEW

**Comment by MarcoFalke** [comment]
Diff to trigger the bug more often:

```diff
diff --git a/src/validationinterface.cpp b/src/validationinterface.cpp
index 613c5b65ef..17c135e7ea 100644
--- a/src/validationinterface.cpp
+++ b/src/validationinterface.cpp
@@ -204,6 +204,7 @@ void CMainSignals::UpdatedBlockTip(const CBlockIndex *pindexNew, const CBlockInd
 
 void CMainSignals::TransactionAddedToMempool(const CTransactionRef& tx, uint64_t mempool_sequence) {
     auto event = [tx, mempool_sequence, this] {
+        UninterruptibleSleep(10ms);
         m_internals->Iterate([&](CValidationInterface& callbacks) { callbacks.TransactionAddedToMempool(tx, mempool_sequence); });
     };
     ENQUEUE_AND_LOG_EVENT(event, "%s: txid=%s wtxid=%s", __func__,
diff --git a/test/functional/wallet_resendwallettransactions.py b/test/functional/wallet_resendwallettransactions.py
index 26df0841d8..0b8f4bf317 100755
--- a/test/functional/wallet_resendwallettransactions.py
+++ b/test/functional/wallet_resendwallettransactions.py
@@ -86,6 +86,7 @@ class ResendWalletTransactionsTest(BitcoinTestFramework):
         child_txid = node.send(outputs=[{addr: 0.5}], options={"inputs": [{"txid":txid, "vout":0}]})["txid"]
         while True:
             txids = node.listreceivedbyaddress(minconf=0, address_filter=addr)[0]["txids"]
+            assert_equal(len(txids), 2)
             if txids == [child_txid, txid]:
                 break
             bumped = node.bumpfee(child_txid)

File: test/functional/wallet_resendwallettransactions.py [hunk 1]
```diff
@@ -89,6 +89,10 @@ def run_test(self):
             if txids == [child_txid, txid]:
                 break
             bumped = node.bumpfee(child_txid)
+            # The scheduler queue creates a copy of the added tx after
+            # send/bumpfee and re-adds it to the wallet (undoing the next
+            # removeprunedfunds). So empty the scheduler queue:
+            node.syncwithvalidationinterfacequeue()
             node.removeprunedfunds(child_txid)
             child_txid = bumped["txid"]
```
Comment (glozow) [inline test/functional/wallet_resendwallettransactions.py:97]: Note to self/reviewer, I think this is why this fix works

Without this, the ordering may be:
1. child1 is created. Added(child1) event is added to queue
2. child2 is created using bumpfee. Removed(child1) and Added(child2) events added to queue
3. removeprunedfunds(child1) deletes child1 from wallet
4. `wallet.transactionAddedToMempool(child1)` actually executes, re-adding child1 to wallet. bug.

And now the ordering must be:
1. child1 is created. Added(child1) event is added to queue
2. `wallet.transactionAddedToMempool(child1)` executes
3. child2 is created using bumpfee. Removed(child1) and Added(child2) events added to queue
4. `wallet.transactionAddedToMempool(child2)` executes, `wallet.transactionRemovedFromMempool(child1)` executes.
5. removeprunedfunds(child1) deletes child1 from wallet

**Review by glozow — COMMENTED** [review]
ACK fa1ce96184a1815f453e64e14d77cb0025800be9

**Review by w0xlt — APPROVED** [review]
ACK https://github.com/bitcoin/bitcoin/commit/fa1ce96184a1815f453e64e14d77cb0025800be9

**Pull request merged**